typescript/web/src/components/workspaces/workspaces.tsx

```diff
@@ -0,0 +1,86 @@
+import { Button, Center, chakra, Flex, Heading, Text } from "@chakra-ui/react";
+import { isEmpty } from "lodash";
+import { FC, useCallback, FormEvent } from "react";
+import NoWorkspacesGraphics from "../graphics/no-workspace";
+import {
+  useWorkspaceNameInput,
+  WorkspaceNameInput,
+  WorkspaceNameInputProvider,
+  WorkspaceNameMessage,
+} from "../workspace-name-input";
+import { useCreateWorkspace } from "../workspace-switcher/create-workspace-modal";
+import {
+  WorkspacesContextProvider,
+  WorkspacesProps,
+} from "./workspaces-context";
+import { WorkspacesList } from "./workspaces-list";
+
+const ChakraNoWorkspaces = chakra(NoWorkspacesGraphics);
+
+const CreateButton: FC<{ isDisabled?: boolean }> = ({ isDisabled }) => (
+  <Button
+    width="full"
+    type="submit"
+    isDisabled={isDisabled}
+    colorScheme="brand"
+    aria-label="create workspace button"
+    mt="8"
+  >
+    Create
+  </Button>
+);
+
+const CreateWorkspaceForm = () => {
+  const { name, error: nameError } = useWorkspaceNameInput();
+  const [create, isLoading, createError] = useCreateWorkspace();
+  const isDisabled = isLoading ? false : !isEmpty(nameError);
+  const empty = name.length === 0;
+
```

Comment:
If in your function you have to separate blocks with new lines, then they are great chances that it has to be factorized 😉

Reply:
Fixed